src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -377,6 +385,9 @@ RouterModifyTaskForShardInterval(Query *originalQuery, ShardInterval *shardInter
 	{
 		RelationRestriction *restriction = lfirst(restrictionCell);
 		List *originalBaserestrictInfo = restriction->relOptInfo->baserestrictinfo;
+		InstantiateQualWalker *instantiateQualWalker =
+			palloc0(sizeof(InstantiateQualWalker));
```

Comment:
I'd add context to the variable/struct name - right now it sounds like that's the walker itself (i.e. the function) not just data.